content/en/blog/_posts/2020-03-16-kubernetes-the-hard-way.md

```diff
@@ -0,0 +1,41 @@
+
+Therefore, the real way to "Learn Kubernetes the Hard Way," is to join the CNCF and get involved in the project itself. And there is only one SIG that could genuinely offer a full-stack learning experience for Kubernetes: SIG Scalability.
+
+The team behind SIG Scalability is responsible for detecting and dealing with issues that arise when Kubernetes clusters are working with upwards of a thousand nodes. Said [Wojiciech Tyczynski](https://github.com/wojtek-t), a staff software engineer at Google and a member of SIG Scalability, the standard size for a test cluster for this SIG is over 5,000 nodes.
+
+And yet, this SIG is not composed of Ph.D.'s in highly scalable systems designs. Many of the folks working with Tyczynski, for example, joined the SIG knowing very little about these types of issues, and often, very little about Kubernetes.
+
+Working on SIG Scalability is like jumping into the deep end of the pool to learn to swim, and the SIG is inherently concerned with the entire Kubernetes project. SIG Scalability focuses on how Kubernetes functions as a whole and at scale. The SIG Scalability team members have an impetus to learn about every system and to understand how all systems interact with one another.
+
+## A complex and rewarding contributor experience
+
+While that may sound complicated (and it is!), that doesn't mean it's outside the reach of an average developer, tester, or administrator. Google software developer Matt Matejczyk has only been on the team since the beginning of 2019, and he's been a valued member of the team since then, ferreting out bugs.
+
+"I am new here," said Matejczyk. "I joined the team in January [2020]. Before that, I worked on AdWords at Google in New York. Why did I join? I knew some [people] there, so that was one of the decisions for me to move. I thought at that time that Kubernetes is a unique, cutting edge technology. I thought it'd be cool to work on that."
+
+Matejczyk was correct about the coolness. "It's cool," he said. "So actually, ramping up on scalability is not easy. There are many things you need to understand. You need to understand Kubernetes very well. It can use every part of Kubernetes. I am still ramping up after these 8 months. I think it took me maybe 3 months to get up to decent speed."
+
+When Matejczyk spoke to what he had worked on during those 8 months, he answered, "Mostly detecting and debugging regressions. This gets really interesting usually, as it takes a lot of time to debug everything from end-to-end, but when you figure out everything, it can be rewarding. What is interesting is that it looks like everything boils down to memory allocation. That is one piece of code, and part of the same binary can affect other code only because it is allocating too much memory."
```

Comment:
I would rephrase the last two sentences. I don't remember what I exactly said, but I'm quite sure I was giving an example and refreezing to a particular bottleneck I was working on during that time. In the current form, a reader may get an impressions that every k8s performance regressions / bottleneck boils down to memory allocation, which is not true.


What about something like this?

> An interesting example is the regression I have been working on recently. We noticed overall slowness of Kubernetes control plane in certain scenarios and we couldn't attribute it to any particular component. In the end, we realized that everything boiled down to the memory allocation on the golang level. It was very counterintuitive to have two completely separate pieces of code (running as a part of the same binary) affecting performance of each other only because one of them was allocating memory too fast. But connecting all the dots and getting to the bottom of regression like this gives a lot of satisfaction.

It's just an example, feel free to rephrase appropriately.